File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/discover/DiscoverMovieListResult.java
package com.crpr.androidcinema.domain.discover;

import com.crpr.androidcinema.domain.common.Result;

import java.util.List;

/**
 * Created by claudioribeiro on 14/07/16.
 */
public class DiscoverMovieListResult extends Result {

    private List<ListMovieModel> _models;

    public DiscoverMovieListResult(int statusCode, List<ListMovieModel> models) {
        super(statusCode);
        this._models = models;
    }

    public DiscoverMovieListResult(int statusCode, String errorMessage) {
        super(statusCode, errorMessage);
    }

    public List<ListMovieModel> getMovies(){
        return _models;
    }
}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/discover/DiscoverMovieListResult.kt
package com.crpr.androidcinema.domain.discover

import com.crpr.androidcinema.domain.common.Result

/**
 * Created by claudioribeiro on 14/07/16.
 */
class DiscoverMovieListResult : Result {

    var movies: List<ListMovieModel>? = null

    constructor(statusCode: Int, models: List<ListMovieModel>?) : super(statusCode) {
        this.movies = models
    }

    constructor(statusCode: Int, errorMessage: String) : super(statusCode, errorMessage) {}
}
